lib/collectinfo_analyzer/show_controller.py

```diff
@@ -281,49 +381,112 @@ def do_dc(self, line):
+                title_every_nth=title_every_nth,
+                flip_output=flip_output,
+                timestamp=timestamp,
+                **self.mods,
+            )
+
+    @CommandHelp(
+        "Displays xdr namespace configuration.",
+        "  Options:",
+        "    -r           - Repeat output table title and row header after every <terminal width> columns.",
+        "                   [default: False, no repetition]",
+        "    --flip      - Flip output table to show Nodes on Y axis and config on X axis.",
+    )
+    def do_namespace(self, line):
+        title_every_nth = util.get_arg_and_delete_from_mods(
+            line=line,
+            arg="-r",
+            return_type=int,
+            default=0,
+            modifiers=self.modifiers,
+            mods=self.mods,
         )
-        node_xdr_build_major_version = 5
 
-        for timestamp in sorted(dc_configs.keys()):
+        flip_output = util.check_arg_and_delete_from_mods(
+            line=line,
+            arg="-flip",
+            default=False,
+            modifiers=self.modifiers,
+            mods=self.mods,
+        ) or util.check_arg_and_delete_from_mods(
+            line=line,
+            arg="--flip",
+            default=False,
+            modifiers=self.modifiers,
+            mods=self.mods,
+        )
+
+        xdr_ns_configs = self.getter.get_xdr_namespaces(for_mods=self.mods["for"])
+
+        for timestamp in xdr_ns_configs.keys():
             cinfo_log = self.log_handler.get_cinfo_log_at(timestamp=timestamp)
-            builds = cinfo_log.get_asd_build()
-            nodes_running_v5_or_higher = False
-            nodes_running_v49_or_lower = False
-
-            for version in builds.values():
-                try:
-                    node_xdr_build_major_version = int(version[0])
-                except Exception:
-                    continue
+            self.view.show_xdr_ns_config(
+                xdr_ns_configs[timestamp],
+                cinfo_log,
+                title_every_nth=title_every_nth,
+                flip_output=flip_output,
+                timestamp=timestamp,
+                **self.mods,
+            )
 
-                if node_xdr_build_major_version >= 5:
-                    nodes_running_v5_or_higher = True
-                else:
-                    nodes_running_v49_or_lower = True
-
-            if nodes_running_v49_or_lower:
-                for dc, configs in dc_configs[timestamp].items():
-                    self.view.show_config(
-                        "%s DC Configuration" % (dc),
-                        configs,
-                        self.log_handler.get_cinfo_log_at(timestamp=timestamp),
-                        title_every_nth=title_every_nth,
-                        flip_output=flip_output,
-                        timestamp=timestamp,
-                        **self.mods
-                    )
+    @CommandHelp(
+        "Displays xdr namespace configuration.",
```

Comment:
Should the help text be about XDR filters?

Reply:
yes